### PR TITLE
Add LSM6DS driver

### LINKED ISSUE #194: bsp: Sailbot: Driver for LSM6DS IMU
This is an issue to develop a driver and a stand-alone testing project for LSM6DS IMU present on SailBot PCB Rev 1.1.1 (1.0).

### PR DESCRIPTION
**Pull request by malishav**
fixes #194

### CODE REVIEW

File: drv/lsm6ds/lsm6ds.c [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+#include <stdio.h>
+
+#include "gpio.h"
+#include "i2c.h"
+#include "lsm6ds.h"
+
+//=========================== defines ==========================================
+
+static const gpio_t scl     = { .port = 1, .pin = 9 };
+static const gpio_t sda     = { .port = 0, .pin = 11 };
```
Comment (aabadie) [inline drv/lsm6ds/lsm6ds.c:21]:
```suggestion
static const gpio_t scl     = { .port = DB_I2C_SCL_PORT, .pin = DB_I2C_SCL_PIN };
static const gpio_t sda     = { .port = DB_I2C_SDA_PORT, .pin = DB_I2C_SDA_PIN };
```
(and run make format to align with the line below)

File: drv/lsm6ds/lsm6ds.c [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+static const gpio_t imu_int = { .port = 0, .pin = 20 };
```
Comment (aabadie) [inline drv/lsm6ds/lsm6ds.c:22]: For compatibility between boards, add a define in the corresponding board config headers. That will avoid pin clash and the driver could also be tested on DK boards.

File: drv/lsm6ds/lsm6ds.c [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+static void cb_imu_int(void *ctx) {
+    (void)ctx;
+    // set data_ready to true
+    _lsm6ds_vars.data_ready = true;
+    //  invoke application callback if initialized
+    if (_lsm6ds_vars.callback != NULL) {
+        _lsm6ds_vars.callback();
+    }
+}
```
Comment (aabadie) [inline drv/lsm6ds/lsm6ds.c:155]: Missing newline at end of files it seems.

**Review by aabadie — CHANGES_REQUESTED** [review]
Some comments, mostly cleanup so rather minor

File: drv/lsm6ds.h [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    int16_t y;  ///< Y axis
+    int16_t z;  ///< Z axis
+} lsm6ds_acc_data_t;
+
+typedef void (*lsm6ds_data_ready_cb_t)(void);  ///< Callback function prototype, it is called on each available sample
+
+void   lsm6ds_init(lsm6ds_data_ready_cb_t callback);
+bool   lsm6ds_data_ready(void);
+void   lsm6ds_read_accelerometer(void);
+int8_t lsm6ds_last_roll(void);
```
[the remark this answers is not in the document]
Comment (malishav) [reply drv/lsm6ds.h:31]: fixed

File: drv/lsm6ds/lsm6ds.c (hunk 1 above)
Comment (malishav) [reply drv/lsm6ds/lsm6ds.c]: fixed

File: projects/01drv_lsm6ds/01drv_lsm6ds.c [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            roll = lsm6ds_last_roll();
+            printf("roll: %d\n", roll);
+        }
+        __WFE();
+    }
+    // one last instruction, doesn't do anything, it's just to have a place to put a breakpoint.
+    __NOP();
+}
+
+//=========================== functions =========================================
```
[the remark this answers is not in the document]
Comment (malishav) [reply projects/01drv_lsm6ds/01drv_lsm6ds.c]: fixed

**Review by aabadie — COMMENTED** [review]
One minor remaining nit pick and after that it's good to go

File: drv/lsm6ds.h [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    int16_t y;  ///< Y axis
+    int16_t z;  ///< Z axis
+} lsm6ds_acc_data_t;
+
+typedef void (*lsm6ds_data_ready_cb_t)(void);  ///< Callback function prototype, it is called on each available sample
+
+/**
+ * @brief Initialize the LSM6DS chip
+ *
+ * @param[in] scl       callback pointer invoked whenever data is ready
```
[the remark this answers is not in the document]
Comment (malishav) [reply drv/lsm6ds.h]: fixed

**Review by aabadie — APPROVED** [review]

**Pull request merged**